metagraph/src/graph/alignment/aligner_aggregator.hpp

```diff
@@ -4,258 +4,229 @@
-                                   a.get_clipping(),
-                                   b.get_score(),
-                                   a.get_sequence().size())
-                < std::make_tuple(b.get_clipping() + b.get_query().size(),
-                                  b.get_clipping(),
-                                  a.get_score(),
-                                  b.get_sequence().size());
-        });
-
-        // recursively construct chains
-        std::string_view this_query = get_query(orientation);
-        std::vector<score_t> best_score(this_query.size() + 1, 0);
-        for (auto it = aln.begin(); it != aln.end(); ++it) {
-            size_t end_pos = it->get_query().data() + it->get_query().size()
-                                - this_query.data();
-            if (it->get_score() > best_score[end_pos]) {
-                best_score[end_pos] = it->get_score();
-                construct_alignment_chain(this_query, Alignment(*it),
-                                          it + 1, aln.end(),
-                                          best_score, push_to_queue);
-            }
+    bool added = false;
+    for (Column c : a->label_columns) {
+        auto &cur_queue = path_queue_[c];
+        if (cur_queue.size() < config_.num_alternative_paths) {
+            cur_queue.emplace(a);
+            added = true;
+        } else if (!cmp_(a, cur_queue.minimum())) {
+            cur_queue.update(cur_queue.begin(), a);
+            added = true;
         }
     }
-}
+    if (!added)
+        return false;
 
-template <class AlignmentCompare>
-inline void AlignmentAggregator<AlignmentCompare>
-::construct_alignment_chain(std::string_view query,
-                            Alignment&& chain,
-                            typename std::vector<Alignment>::iterator begin,
-                            typename std::vector<Alignment>::iterator end,
-                            std::vector<score_t> &best_score,
-                            const std::function<void(Alignment&&)> &callback) {
-    assert(begin <= end);
-    assert(chain.size());
-
-    const char *chain_begin = chain.get_query().data();
-    const char *chain_end = chain_begin + chain.get_query().size();
-    if (begin == end || chain_end == query.data() + query.size()) {
-        callback(std::move(chain));
-        return;
+    if (!cmp_(a, nqueue.minimum())) {
+        nqueue.update(nqueue.begin(), a);
+    } else {
+        nqueue.emplace(a);
     }
+    return true;
+}
 
-    size_t k = graph_.get_k();
-    score_t score = chain.get_score();
-
-    bool called = false;
-    for (auto it = begin; it != end; ++it) {
-        if (it->get_offset())
-            continue;
+template <class AlignmentCompare>
+inline auto AlignmentAggregator<AlignmentCompare>
+::get_min_path_score(const Vector<Column> &labels) const -> score_t {
+    score_t global_min = get_max_path_score() * config_.rel_score_cutoff;
```

Comment:
You're using `global_min` here but not in `get_min_path_score(Column label)`, so the results for `get_min_path_score(t)` and `get_min_path_score({ t })` might be different.

Am I missing something?